src/components/Verse/OverflowVerseActionsMenuBody.tsx

```diff
@@ -169,41 +169,50 @@ const OverflowVerseActionsMenuBody: React.FC<Props> = ({
     );
 
     if (isLoggedIn()) {
+      // optimistic update, we are making assumption that the bookmark update will succeed
+      mutate((currentIsVerseBookmarked) => !currentIsVerseBookmarked, {
```

Comment:
optimisticly update the state. So psychologically, user think it's done. 
So he won't try to press multiple time. It's fire and forget. Peace in mind for the users